client/systems.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"golang.org/x/xerrors"
+
+	"github.com/snapcore/snapd/snap"
+)
+
+// SystemModelData contains information about the model
+type SystemModelData struct {
+	// Model as the model assertion
+	Model string `json:"model,omitempty"`
+	// BrandID corresponds to brand-id in the model assertion
+	BrandID string `json:"brand-id,omitempty"`
+	// DisplayName is human friendly name, corresponds to display-name in
+	// the model assertion
+	DisplayName string `json:"display-name,omitempty"`
+}
+
+type System struct {
+	// Current is true when the system running now was installed from that
+	// recovery seed
+	Current bool `json:"current,omitempty"`
+	// Label of the recovery system
+	Label string `json:"label,omitempty"`
+	// Model information
+	Model SystemModelData `json:"model,omitempty"`
+	// Brand information
+	Brand snap.StoreAccount `json:"brand,omitempty"`
+	// Actions available for this system
+	Actions []SystemAction `json:"actions,omitempty"`
+}
+
+type SystemAction struct {
+	// ID used when referencing given action in recover requests
+	ID string `json:"id,omitempty"`
```

Comment:
let's leave out ID until we look how to request an action really works,   I actually expect that a first iteration will use just label and mode